src/sphexa/sphexa.cpp

```diff
@@ -114,29 +110,24 @@ int main(int argc, char** argv)
     viz::init_catalyst(argc, argv);
     viz::init_ascent(d, domain.startIndex());
 
-    Propagator propagator(ngmax, ng0, output, d.rank);
-
-    MasterProcessTimer totalTimer(output, d.rank);
+    MasterProcessTimer totalTimer(output, rank);
     totalTimer.start();
-    for (d.iteration = 0; d.iteration <= maxStep; d.iteration++)
+    int startIteration = d.iteration;
+    for (; d.iteration <= maxStep; d.iteration++)
     {
-        if (ve) { propagator.hydroStepVE(domain, d); }
-        else
-        {
-            propagator.hydroStep(domain, d);
-        }
+        propagator->step(domain, d);
 
         fileutils::writeColumns(constantsFile, ' ', d.iteration, d.ttot, d.minDt, d.etot, d.ecin, d.eint, d.egrav);
 
         if ((writeFrequency > 0 && d.iteration % writeFrequency == 0) || writeFrequency == 0)
         {
-            fileWriter->dump(d, domain.startIndex(), domain.endIndex(), outDirectory + "dump_sedov");
+            fileWriter->dump(d, domain.startIndex(), domain.endIndex(), box, outFile);
         }
 
         if (d.iteration % 5 == 0) { viz::execute(d, domain.startIndex(), domain.endIndex()); }
     }
 
-    totalTimer.step("Total execution time of " + std::to_string(maxStep) + " iterations of Sedov");
+    totalTimer.step("Total execution time of " + std::to_string(maxStep - startIteration + 1) + " iterations of Sedov");
```

Comment:
you may want to update the message when running `noh` instead of sedov ?

Reply:
done